### PR TITLE
fix(custom-resources): correctly handle booleans conversion

### LINKED ISSUE #3933: Firelens taskDefinition with custom-resources got "logConfiguration.options['auto_create_group'] to be a string" error
## :question: General Issue

I was trying to create a Fargate task definition with the Firelens support with `@aws-cdk/custom-resources`. When I deploy it I got the following error:

```
Failed to update resource. Expected params.containerDefinitions[1].logConfiguration.options['auto_create_group'] to be a string
```

I am not sure if this is a bug related to AWS CDK or AWS Javascript SDK.

### The Question

```js
import cdk = require('@aws-cdk/core');
import { Vpc } from '@aws-cdk/aws-ec2';
import ecs = require('@aws-cdk/aws-ecs');
import customresources = require('@aws-cdk/custom-resources');

export class AwsFireLensStack extends cdk.Stack {
  public taskArn: string
  public FirelensEnabledTask: any

  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const vpc = Vpc.fromLookup(this, 'VPC', {
      isDefault: true
    })

    const cluster = new ecs.Cluster(this, 'Cluster', {
      vpc: vpc
    })

    this.FirelensEnabledTask = new customresources.AwsCustomResource(this, 'FirelensEnabledTask', {
      onUpdate: { // will also be called for a CREATE event
        service: 'ECS',
        action: 'registerTaskDefinition',
        parameters: {
          "requiresCompatibilities": ["FARGATE"],
          "networkMode": "awsvpc",
          "family": "firelens-demo",
          "cpu": "1024",
          "memory": "2048",
          "containerDefinitions": [
            {
              "essential": true,
              "image": "amazon/aws-for-fluent-bit:latest",
              "name": "log_router",
              "firelensConfiguration": {
                "type": "fluentbit"
              },
              "memory": 25
            },
            {
              "essential": true,
              "image": "nginx",
              "name": "app",
              "portMappings": [
                {
                  "hostPort": 80,
                  "protocol": "tcp",
                  "containerPort": 80
                }],
              "logConfiguration": {
                "logDriver": "awsfirelens",
                "options": {
                  "Name": "cloudwatch",
                  "region": `${this.region}`,
                  "auto_create_group": "true",
                  "log_group_name": "firelens-testing-fluent-bit",
                  "log_stream_prefix": "from-fluent-bit"
                }
              },
              "memory": 100
            }
          ]
        },
        physicalResourceId: Date.now().toString()
      },

    })
    new cdk.CfnOutput(this, 'NewTaskArn', {
      value: this.FirelensEnabledTask.getData('taskDefinition.taskDefinitionArn')
    })
  }
}
```
when I `cdk deploy` I got this error:

圖片

If I check the custom resource in cloudformation generated from `cdk synth` I got `auto_create_group: "true"` in the YAML. And it looks like the custom-resources provider fails to validate it as a string.

```yaml
  FirelensEnabledTask266C2132:
    Type: Custom::AWS
    Properties:
      ServiceToken:
        Fn::GetAtt:
          - AWS679f53fac002430cb0da5b7982bd22872D164C4C
          - Arn
      Create:
        service: ECS
        action: registerTaskDefinition
        parameters:
          requiresCompatibilities:
            - FARGATE
          networkMode: awsvpc
          family: firelens-demo
          cpu: "1024"
          memory: "2048"
          containerDefinitions:
            - essential: true
              image: amazon/aws-for-fluent-bit:latest
              name: log_router
              firelensConfiguration:
                type: fluentbit
              memory: 25
            - essential: true
              image: nginx
              name: app
              portMappings:
                - hostPort: 80
                  protocol: tcp
                  containerPort: 80
              logConfiguration:
                logDriver: awsfirelens
                options:
                  Name: cloudwatch
                  region: ap-northeast-1
                  auto_create_group: "true"
                  log_group_name: firelens-testing-fluent-bit
                  log_stream_prefix: from-fluent-bit
              memory: 100
        physicalResourceId: "1567598358544"
      Update:
        service: ECS
        action: registerTaskDefinition
        parameters:
          requiresCompatibilities:
            - FARGATE
          networkMode: awsvpc
          family: firelens-demo
          cpu: "1024"
          memory: "2048"
          containerDefinitions:
            - essential: true
              image: amazon/aws-for-fluent-bit:latest
              name: log_router
              firelensConfiguration:
                type: fluentbit
              memory: 25
            - essential: true
              image: nginx
              name: app
              portMappings:
                - hostPort: 80
                  protocol: tcp
                  containerPort: 80
              logConfiguration:
                logDriver: awsfirelens
                options:
                  Name: cloudwatch
                  region: ap-northeast-1
                  auto_create_group: "true"
                  log_group_name: firelens-testing-fluent-bit
                  log_stream_prefix: from-fluent-bit
              memory: 100
        physicalResourceId: "1567598358544"
    UpdateReplacePolicy: Delete
    DeletionPolicy: Delete
    Metadata:
      aws:cdk:path: awsFireLensDemo1/FirelensEnabledTask/Resource/Default
```

### Environment

  - **CDK CLI Version:** 1.6.1 (build a09203a)
  - **Module Version:** 1.6.1
  - **OS:** OSX Mojave
  - **Language:** TypeScript

### Other information

My Firelens sample configuration is from [here](https://github.com/aws/containers-roadmap/blob/7bda4999601c66aa024c96c6ad95ffba60aed0b1/preview-programs/firelens/cloudwatch_task_definition.json#L23)

### PR DESCRIPTION
**Pull request by jogold**
Encode booleans as special strings in the construct and decode them in the Lambda.

Allows to discriminate between real booleans (`true`/`false`) and booleans expressed
as strings (`'true'`/`'false'`)

Fixes #3933

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jogold** [comment]
@eladb just pushed a commit to fix the header level in the README. Can you re-approve? Thanks.

File: packages/@aws-cdk/custom-resources/README.md [hunk 1]
```diff
@@ -101,3 +101,6 @@ const awsCustom2 = new AwsCustomResource(this, 'API2', {
   }
 })
 ```
+
+##### Booleans conversion
+Since custom resource properties are always stringified by CloudFormation, `true`/`false` values will automatically be converted back to booleans. If the strings `'true'`/`'false'` are required in the SDK call, use `'true!'`/`'false!'`.
```
Comment (eladb) [inline packages/@aws-cdk/custom-resources/README.md]: Wait, can’t we convert Boolean true/false to their encoded values in the construct?

Comment (jogold) [reply packages/@aws-cdk/custom-resources/README.md]: Passing either `true`/`false` or `'true'`/`'false'` in the custom resource properties will end up as a string in the event received by the Lambda function. This conversions needs to be handled in the Lambda.

Comment (jogold) [reply packages/@aws-cdk/custom-resources/README.md]: OK, you mean the other way around? Let the user specify "normal" values and encode them in the construct as special strings to be able to decode in the Lambda?

**Pull request merged**